parts/22-query.sh

```diff
@@ -952,6 +952,47 @@ query_tool-last-used-date() { ## : When was the most recent invocation of every
 	EOF
 }
 
+local_query-toolusebygroup() { ## : <year-month> <group> Lists count of tools used by all users in a group
+        handle_help "$@" <<-EOFhelp
+                Lists tools use count by users in group.
+                Requires <year-month> (2022-03) and <group> 
+
+                Example:
+                $ gxadmin local query-toolusebygroup 2022-02 CSIRO
+                tool_id                                             |             username             | count 
+                ----------------------------------------------------+----------------------------------+-------
+                CONVERTER_gz_to_uncompressed                        | erin_hahn                        |     1
+                Convert characters1                                 | Benson, Derek (IM&T, Pullenvale) |     1
+                Cut1                                                | Benson, Derek (IM&T, Pullenvale) |     1
+                Cut1                                                | jodie.vandekamp_at_csiro.au      |     1
+
+
+EOFhelp
+
+if (( $# > 0 )); then
+                wheredate=" date_trunc('month', job.create_time) = '$1-01'"
+                wheregroup="'$2'"
+fi
+        read -r -d '' QUERY <<-EOF
```

Comment:
```suggestion
        username=$(gdpr_safe galaxy_user.username username "Anonymous User")
        read -r -d '' QUERY <<-EOF
```

For queries that obtain user information, we have a function to help us keep to GDPR compliance and hide unnecessary usernames. While not relevant for your case (judging by the function documentation) it is for the rest of our users especially if they want to share results with e.g. colleagues.

Reply:
Resolved as suggested.